### PR TITLE
[Plot] - unionized removeDataset

### PR DESCRIPTION
**Pull request by bluong**
After some muddling around, removeDataset now is union typed

### CODE REVIEW

File: plottable.js [hunk 1]
```diff
@@ -6425,16 +6425,32 @@ var Plottable;
                 }
                 return this;
             };
-            AbstractPlot.prototype.removeDataset = function (datasetOrKeyOrArray) {
+            /**
+             * Removes a dataset by the given identifier
+             *
+             * @param {string | Dataset | any[]} datasetIdentifer The identifier for the dataset to remove
+             * If a string is inputted, it is interpreted as the dataset key to remove.
```
Comment (jtlan) [inline plottable.js]: "as the key of the Dataset to remove" ?

File: plottable.js [hunk 2]
```diff
@@ -6425,16 +6425,32 @@ var Plottable;
...
+             * If a Dataset is inputted, the first Dataset in the plot that is the same will be removed.
+             * If an any[] is inputted, the first data array in the plot that is the same will be removed.
```
Comment (jtlan) [inline plottable.js]: Generally, maybe "If ____, [...]" is clearer than "If a ______ is inputted, [...]"

File: src/components/plots/abstractPlot.ts [hunk 3]
```diff
@@ -311,37 +311,34 @@ export module Plot {
     }
 
     /**
-     * Removes a dataset by string key
+     * Removes a dataset by the given identifier
      *
-     * @param {string} key The key of the dataset
-     * @return {Plot} The calling Plot.
+     * @param {string | Dataset | any[]} datasetIdentifer The identifier for the dataset to remove
```
Comment (jtlan) [inline src/components/plots/abstractPlot.ts]: "as the key of the Dataset to remove" ?

File: src/components/plots/abstractPlot.ts [hunk 4]
```diff
@@ -311,37 +311,34 @@ export module Plot {
...
+     * If a string is inputted, it is interpreted as the dataset key to remove.
+     * If a Dataset is inputted, the first Dataset in the plot that is the same will be removed.
+     * If an any[] is inputted, the first data array in the plot that is the same will be removed.
```
Comment (jtlan) [inline src/components/plots/abstractPlot.ts]: Generally, maybe "If ____, [...]" is clearer than "If a ______ is inputted, [...]"

File: src/components/plots/abstractPlot.ts [hunk 5]
```diff
@@ -311,37 +311,34 @@ export module Plot {
...
-          key = this._datasetKeysInOrder[idx];
+      if (typeof(datasetIdentifier) === "string") {
+        key = <string> datasetIdentifier;
+      } else if (typeof (datasetIdentifier) === "object") {
+
+        var index = -1;
+        if (typeof ((<Dataset> datasetIdentifier).data) !== "undefined") {
+          var datasetArray = this.datasets();
+          index = datasetArray.indexOf(<Dataset> datasetIdentifier);
+        } else if (typeof ((<any[]> datasetIdentifier).map) !== "undefined") {
```
Comment (jtlan) [inline src/components/plots/abstractPlot.ts]: Why the switch away from `instanceof`?

**Comment by jtlan** [comment]
Questions, and some docs stuff.

File: src/components/plots/abstractPlot.ts (hunk 5 above)
Comment (bluong) [inline src/components/plots/abstractPlot.ts]: instanceof requires an any as a first argument or so (I could any cast it I guess...).

Also, I was nervous about https://github.com/palantir/plottable/issues/638

**Pull request merged**